frame/staking/src/lib.rs

```diff
@@ -1236,6 +1294,57 @@ decl_module! {
 			<Self as Store>::UnappliedSlashes::insert(&era, &unapplied);
 		}
 
+		/// Make one nominator's payout for one era.
+		///
+		/// - `who` is the controller account of the nominator to pay out.
+		/// - `era` may not be lower than one following the most recently paid era. If it is higher,
+		///   then it indicates an instruction to skip the payout of all previous eras.
+		/// - `validators` is the list of all validators that `who` had exposure to during `era`.
+		///   If it is incomplete, then less than the full reward will be paid out.
+		///
+		/// WARNING: once an era is payed for a validator such validator can't claim the payout of
+		/// previous era.
+		///
+		/// WARNING: Incorrect arguments here can result in loss of payout. Be very careful.
+		///
+		/// # <weight>
+		/// - Number of storage read of `O(validators)`; `validators` is the argument of the call.
+		/// - Each storage read is `O(N)` size and decode complexity; `N` is the  maximum
+		///   nominations that can be given to a single validator.
+		/// - Computation complexity: `O(MAX_NOMINATIONS * logN)`; `MAX_NOMINATIONS` is the
+		///   maximum number of validators that may be nominated by a single nominator, it is
+		///   bounded only economically (all nominators are required to place a minimum stake).
+		/// # </weight>
+		#[weight = SimpleDispatchInfo::FixedNormal(500_000)]
+		// TODO: Limit the amount of nominators that can be assigned to a validator by Phragmen.
```

Comment:
should we also limit this in the current PR ?

Reply:
Also if decoding the structure Exposure which contains all nominations for the validator is to costy during this reward we could try to store on chain the nominations of a nominator so it only query one structure for himself and not one structure for each validator it has nominated

Reply:
that would likely take a lot of state and i/o at the phragmen time, so far from ideal. better to just limit the number of nominators at the phragmen stage.

Reply:
we can introduce per-validator indices to denote which exposure index our nominator is for each validator: switch the parameter here to `fn payout_nominator(origin, era: EraIndex, validators: Vec<(T::AccountId, u32)>)` where the `u32` is the index in to the validator's exposure list.

we should still limit the number of nominators allowed for each validator to something sensible (e.g. 64) just to avoid a quadratic attack on decoding (e.g. a validator with 10,000 tiny nominators, thenn calling this function for each one would result in `O(10,000**2)` with only 10,000 dispatches paid.

Reply:
number of nominators is fixed for nominator reward using exposureclipped principle.

payout_nominator using nominator in validator exposure is done.